src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -214,6 +263,19 @@ public void subBatchExecute(List<IngestDocumentWrapper> ingestDocumentWrappers,
         });
     }
 
+    @Override
+    public void batchExecute(List<IngestDocumentWrapper> ingestDocumentWrappers, Consumer<List<IngestDocumentWrapper>> handler) {
+        super.batchExecute(ingestDocumentWrappers, handler);
+    }
+
+    protected List<List<IngestDocumentWrapper>> cutBatches(List<IngestDocumentWrapper> ingestDocumentWrappers) {
+        List<List<IngestDocumentWrapper>> batches = new ArrayList();
```

Comment:
If we know the size of this list upfront, better to specify it.
```suggestion
        List<List<IngestDocumentWrapper>> batches = new ArrayList((ingestDocumentWrappers.size() + this.batchSize) / this.batchSize - 1);
```